src/rules/property-unit-whitelist/index.js

```diff
@@ -0,0 +1,45 @@
+import { isObject } from "lodash"
+import valueParser from "postcss-value-parser"
+
+import {
+  declarationValueIndexOffset,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "property-unit-whitelist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (p, u) => `Unexpected unit "${u}" for property "${p}"`,
+})
+
+export default function (whitelist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: whitelist,
+      possible: [isObject],
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+
+      const { prop, value } = decl
+      const whitelistInProp = whitelist[prop]
+
```

Comment:
This variable should be named `propWhitelist`.

And at this point you can early return if it is undefined, right?

```js
const propWhitelist = whitelist[prop]
if (!propWhitelist) { return }
```